Scripts/Runtime/Data/VirtualData/VDContextID.cs

```diff
@@ -0,0 +1,66 @@
+using System;
+using Unity.Collections;
+using Unity.Entities;
+
+namespace Anvil.Unity.DOTS.Data
+{
+    internal readonly struct VDContextID : IEquatable<VDContextID>
+    {
+        public static bool operator==(VDContextID lhs, VDContextID rhs)
+        {
+            return lhs.Entity == rhs.Entity && lhs.Context == rhs.Context;
+        }
+
+        public static bool operator!=(VDContextID lhs, VDContextID rhs)
+        {
+            return !(lhs == rhs);
+        }
+        
+        public readonly Entity Entity;
+        public readonly uint Context;
+        
+        internal VDContextID(Entity entity, uint context)
+        {
+            Entity = entity;
+            Context = context;
+        }
+        
+        public bool Equals(VDContextID other)
+        {
+            return Entity == other.Entity && Context == other.Context;
+        }
+
+        public override bool Equals(object compare)
+        {
+            return compare is VDContextID id && Equals(id);
+        }
+
+        public override int GetHashCode()
```

Comment:
Yeah essentially it's better for us to use it explicitly because we have the benefit of knowing which values matter. 

`Entity.GetHashCode` just returns the `Entity.Index` so we can shortcut the method call. 

Using the `ValueTuple` implementation gives us a very good chance of not having collisions which speeds things up by not needing to have the equality check.